source/common/tls/context_impl.cc

```diff
@@ -710,7 +710,9 @@ ClientContextImpl::newSsl(const Network::TransportSocketOptionsConstSharedPtr& o
                                                  : server_name_indication_;
   if (!server_name_indication.empty()) {
     const int rc = SSL_set_tlsext_host_name(ssl_con.get(), server_name_indication.c_str());
-    RELEASE_ASSERT(rc, Utility::getLastCryptoError().value_or(""));
+    if (rc != 1) {
+      return nullptr;
+    }
```

Comment:
I think this will be very confusing to debug if this is happening to people. Can we leave a TODO to somehow add a response code / message specific to this case? Possibly could do a quick WARN_EVERY log message as a stop gap. Up to you.